### PR TITLE
nixos/update-users-groups: fix cross compilation

### PR DESCRIPTION
**Pull request by r-vdp**
## Description of changes

Since #246772, cross compiled NixOS is broken because the DateTime perl
package that was used in the update-users-groups.pl script depends on
Testutf8 which does not cross compile (see #198548).

This PR drops the DateTime dependency in favour of TimePiece, which has
less dependencies and whose closure does cross compile.

Ping @lopsided98 @RaitoBezarius

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
Let's test it carefully.

**Comment by R-VdP** [comment]
Yeah. I don't do cross-compiled NixOS, so it would be useful if someone who does could confirm that this works now.

**Review by lopsided98 — APPROVED** [review]
This fixes building my cross-compiled NixOS systems. I also deployed this change to one of these systems without any problems, although I don't use the account expiration feature.

File: nixos/modules/config/users-groups.nix [hunk 1]
```diff
@@ -648,7 +648,7 @@ in {
         install -m 0700 -d /root
         install -m 0755 -d /home
 
-        ${pkgs.perl.withPackages (p: [ p.FileSlurp p.JSON p.DateTime ])}/bin/perl \
+        ${pkgs.perl.withPackages (p: [ p.FileSlurp p.JSON p.TimePiece ])}/bin/perl \
```
Comment (Mic92) [inline nixos/modules/config/users-groups.nix:651]: I don't even need a module for that, it's in core: https://perldoc.perl.org/Time::Piece

```
use Time::Piece;

my $t = localtime;
print "Time is $t\n";
print "Year is ", $t->year, "\n";
```

Comment (Mic92) [reply nixos/modules/config/users-groups.nix:651]:
```suggestion
        ${pkgs.perl.withPackages (p: [ p.FileSlurp p.JSON ])}/bin/perl \
```

**Comment by R-VdP** [comment]
@ofBorg test user-expiry

**Comment by SuperSandro2000** [comment]
> This PR drops the DateTime dependency in favour of TimePiece, which has
> less dependencies and whose closure does cross compile.

That's great. I already wanted to complain about those new depedencies:

```
perl: +37.8 KiB
perl5.38.0-B-Hooks-EndOfScope: ∅ → 0.24, +19.2 KiB
perl5.38.0-CPAN-Meta-Check: ∅ → 0.014
perl5.38.0-Class-Data-Inheritable: ∅ → 0.08
perl5.38.0-Class-Inspector: ∅ → 1.36, +24.3 KiB
perl5.38.0-Class-Singleton: ∅ → 1.6, +15.4 KiB
perl5.38.0-DateTime: ∅ → 1.54, +205.6 KiB
perl5.38.0-DateTime-Locale: ∅ → 1.28, +14837.5 KiB
perl5.38.0-DateTime-TimeZone: ∅ → 2.44, +6846.5 KiB
perl5.38.0-Devel-StackTrace: ∅ → 2.04, +26.5 KiB
perl5.38.0-Dist-CheckConflicts: ∅ → 0.11, +13.0 KiB
perl5.38.0-Eval-Closure: ∅ → 0.14, +12.7 KiB
perl5.38.0-Exception-Class: ∅ → 1.44, +36.8 KiB
perl5.38.0-File-ShareDir: ∅ → 1.118, +23.9 KiB
perl5.38.0-MRO-Compat: ∅ → 0.13, +13.7 KiB
perl5.38.0-Module-Implementation: ∅ → 0.09, +11.4 KiB
perl5.38.0-Module-Runtime: ∅ → 0.016, +19.3 KiB
perl5.38.0-Package-Stash: ∅ → 0.39, +29.1 KiB
perl5.38.0-Params-ValidationCompiler: ∅ → 0.30, +44.4 KiB
perl5.38.0-Role-Tiny: ∅ → 2.001004, +26.5 KiB
perl5.38.0-Specio: ∅ → 0.46, +272.2 KiB
perl5.38.0-Sub-Exporter-Progressive: ∅ → 0.001013, +8.2 KiB
perl5.38.0-Sub-Identify: ∅ → 0.14, +23.4 KiB
perl5.38.0-Sub-Quote: ∅ → 2.006006, +31.2 KiB
perl5.38.0-Test-Requires: ∅ → 0.11
perl5.38.0-namespace-autoclean: ∅ → 0.29, +15.4 KiB
perl5.38.0-namespace-clean: ∅ → 0.27, +19.8 KiB
```

**Review by SuperSandro2000 — APPROVED** [review]

**Review by tpwrules — APPROVED** [review]
Confirmed this fixes my cross-compilation issues too. Please merge.

**Comment by Mic92** [comment]
Tested for a bit and works.

**Pull request merged**